hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/BalanceChange.java

```diff
@@ -198,13 +202,20 @@ public void replaceNonEmptyAliasWith(final EntityNum createdId) {
         if (isAlias(accountId)) {
             accountId = createdId.toGrpcAccountId();
             account = Id.fromGrpcAccount(accountId);
+            aliasToNewId = Pair.of(alias, accountId);
             alias = ByteString.EMPTY;
         } else if (hasNonEmptyCounterPartyAlias()) {
             counterPartyAccountId = createdId.toGrpcAccountId();
+            aliasToNewId = Pair.of(counterPartyAlias, accountId);
```

Comment:
⚠️ Shouldn't this be,
```
aliasToNewId = Pair.of(counterPartyAlias, counterPartyAccountId);
```
?

Reply:
Yup, probably missed that when copy-pasting... Changed the whole approach, though, [as described here](https://github.com/hashgraph/hedera-services/pull/4306#discussion_r1026363960)